source/tutorials/kubernetes.rst

```diff
@@ -3,170 +3,340 @@
+      sudo systemctl mask docker-set-default-runtime.service
+
+#. Once the cluster initialization is complete, continue reading about how to
+   `Use your cluster`_.
+
+containerd
+----------
+
+`containerd <https://containerd.io/>`_ is the runtime  that the Docker engine
+is built ontop of. Kubernetes can use containerd directly instead of going
+through the Docker engine for increased robustness and performance. See the
+`blog post on kubernetes containerd integration
+<https://kubernetes.io/blog/2018/05/24/kubernetes-containerd-integration-goes-ga/>`_
+for more details.
+
+#. Start the containerd service and enable it to run at boot automatically:
 
    .. code-block:: bash
 
-      sudo -s
+      sudo systemctl enable --now containerd.service
 
-#.  Create (or edit if it exists) the hosts file that Kubernetes will read to
-    locate the master's host:
 
-    .. code-block:: bash
+#. Configure kubelet to use containerd and reload the service.
 
-       echo "127.0.0.1 localhost `hostname`" >> /etc/hosts
+   .. code-block:: bash
 
-#.  Exit root:
+      sudo mkdir -p  /etc/systemd/system/kubelet.service.d/
 
-    .. code-block:: bash
+      cat << EOF | sudo tee  /etc/systemd/system/kubelet.service.d/0-containerd.conf
+      [Service]                                                 
+      Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --runtime-request-timeout=15m --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+      EOF
 
-       exit
+   .. code:: bash
 
-Configure and run Kubernetes
-****************************
+      systemctl daemon-reload
 
-This section describes how to configure and run Kubernetes with CRI-O and kata-runtime. We will be using  the :command:`kubectl` command-line interface to control Kubernetes.
+#. Run the kubeadm command to initialize the master node with the
+   :command:`--cri-socket` parameter:
 
-Configure and run CRI-O + kata-runtime
-======================================
+   .. important:: 
 
-#.  Enable the CRI-O service:
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. In this example, *flannel* is
+      being used for the pod networking. See `Choose a pod network add-on`_
+      for more information.
 
-    .. code-block:: bash
+   .. code-block:: bash
 
-       sudo systemctl enable crio.service
+      sudo kubeadm init \
+      --cri-socket=/run/containerd/containerd.sock \ #required for containerd
+      --pod-network-cidr 10.244.0.0/16   #required for flannel
 
-#.  Enter these commands to restart the system services so that CRI-O will be restarted, and will automatically start with every reboot.
 
-    .. code-block:: bash
+#. (Optional) By default, containerd will use the default runc as the default
+   runtime. The with or without `Kata Containers`_. runtime can be set
+   per-pod. See the Kata Containers documentation on `creating an untrusted
```

Comment:
I just confirmed the Kata documentation is old and out of date.  The containerd documentation should mention runtime class as well and describe how to do it that way.  I think you could still link to it but remove mentions of trusted/untrusted so that when that doc is updated these instructions still work. I *think* that still works by using it as trusted/untrusted but it is not the official way to do things anymore.

Reply:
Thanks. Can you post the link showing how to set the RuntimeClass?

Reply:
Nope. It would be similar to how crio does it but I have not done that yet.  You would have to look at how crio does it and refer to the kubernetes doc on runtime class for the difference.  I can do this in the Kata doc some time after KubeCon but it will take me time to also figure it out.  For this doc you can just refer to the Kata repo as that file would stay the same and just be updated.